lib/CAPI/Dialect/HW.cpp

```diff
@@ -183,6 +195,24 @@ MLIR_CAPI_EXPORTED MlirAttribute hwParamDeclAttrGetValue(MlirAttribute decl) {
   return wrap(unwrap(decl).cast<ParamDeclAttr>().getValue());
 }
 
+MLIR_CAPI_EXPORTED bool hwAttrIsAParamDeclRefAttr(MlirAttribute attr) {
+  return unwrap(attr).isa<ParamDeclRefAttr>();
+}
+
+MLIR_CAPI_EXPORTED MlirAttribute hwParamDeclRefAttrGet(MlirContext ctx,
+                                                       MlirStringRef cName) {
```

Comment:
Another design point to consider is accepting a `StringAttr` instead of a stringref. Could have some advantages (IIRC consistency), but it's your call.

Reply:
As in, a `MlirAttribute` that is cast to a `StringAttr`? would allow for the `ctx` parameter to be removed, so i guess that would be a fair choice.

Reply:
at least on the C++ side, the inferred context builds are much nicer to use.  But it often is nice to have both so you aren't just moving the work to making a full stringattr, with it's builder needing a context, just to avoid passing a context.